src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11180,117 +11185,118 @@ private ExpressionSyntax ParseExpressionContinued(ExpressionSyntax leftOperand,
                         rhs = this.ParseSubExpression(newPrecedence);
                     }
 
-                    leftOperand = _syntaxFactory.AssignmentExpression(opKind, leftOperand, opToken, rhs);
+                    leftOperand = _syntaxFactory.AssignmentExpression(opKind, leftOperand, operatorToken, rhs);
                 }
-                else if (opKind == SyntaxKind.SwitchExpression)
+                else if (IsExpectedBinaryOperator(tk))
                 {
-                    leftOperand = ParseSwitchExpression(leftOperand, opToken);
+                    leftOperand = _syntaxFactory.BinaryExpression(opKind, leftOperand, operatorToken, this.ParseSubExpression(newPrecedence));
                 }
-                else if (opKind == SyntaxKind.WithExpression)
+                else
                 {
-                    leftOperand = ParseWithExpression(leftOperand, opToken);
+                    throw ExceptionUtilities.Unreachable();
                 }
-                else if (tk == SyntaxKind.DotDotToken)
+
+                return true;
+            }
+
+            SyntaxToken eatOperatorToken(SyntaxKind operatorKind)
+            {
+                // Combine tokens into a single token if needed
+                if (operatorKind is SyntaxKind.RightShiftExpression or SyntaxKind.RightShiftAssignmentExpression)
                 {
-                    // Operator ".." here can either be a binary or a postfix unary operator:
-                    Debug.Assert(opKind == SyntaxKind.RangeExpression);
+                    // >>  or >>=
+                    return SyntaxFactory.Token(
+                        this.EatToken().GetLeadingTrivia(),
+                        operatorKind == SyntaxKind.RightShiftExpression ? SyntaxKind.GreaterThanGreaterThanToken : SyntaxKind.GreaterThanGreaterThanEqualsToken,
```

Comment:
It feels like we are translating the operator-token-kind into an expression-kind in a previous step, and then here doing the same translation in the other direction. Would it be simpler to just pass along the operator-token-kind? It's possible I am missing something.

Reply:
Let me see about that!

Reply:
This did improve things.  Thanks!